src/frontend/screens/ProjectInviteModal/index.tsx

```diff
@@ -0,0 +1,273 @@
+import * as React from "react";
+import { Alert, StyleSheet, View } from "react-native";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import { FormattedMessage, defineMessages, useIntl } from "react-intl";
+import { BottomSheetModal } from "@gorhom/bottom-sheet";
+
+import { MAPEO_BLUE, WHITE } from "../../lib/styles";
+import BottomSheet from "../../sharedComponents/BottomSheet";
+import Text from "../../sharedComponents/Text";
+import { DoneIcon } from "../../sharedComponents/icons";
+import Circle from "../../sharedComponents/icons/Circle";
+import Loading from "../../sharedComponents/Loading";
+import ModalActionButtons from "./ModalActionButtons";
+
+interface ProjectInviteDetails {
+  project: {
+    id: string;
+    name: string;
+  };
+  role: string;
+}
+
+type LoadingStatus = {
+  type: "loading";
+};
+
+type ErrorStatus = {
+  type: "error";
+  info: { error: Error };
+};
+
+type SuccessStatus = {
+  type: "success";
+  info: {
+    inviteDetails: ProjectInviteDetails;
+  };
+};
+
+const m = defineMessages({
+  title: {
+    id: "screens.ProjectInviteModal.title",
+    defaultMessage: "Join {projectName}",
+  },
+  inviteMessage: {
+    id: "screens.ProjectInviteModal.inviteMessage",
+    defaultMessage: "You've been invited to join\n {projectName} as a {role}",
+  },
+  declineInvite: {
+    id: "screens.ProjectInviteModal.declineInvite",
+    defaultMessage: "Decline Invite",
+  },
+  joinAndSync: {
+    id: "screens.ProjectInviteModal.joinAndSync",
+    defaultMessage: "Join And Sync",
+  },
+  close: {
+    id: "screens.ProjectInviteModal.close",
+    defaultMessage: "Close",
+  },
+  tryAgain: {
+    id: "screens.ProjectInviteModal.tryAgain",
+    defaultMessage: "Try Again",
+  },
+  noInviteFound: {
+    id: "screens.ProjectInviteModal.noInviteFound",
+    defaultMessage: "No Invite Found",
+  },
+});
+
+// Gregor TODO: Add method to API for decrypting invite
+const getProjectInviteDetails = async (
+  inviteKey: string
+): Promise<ProjectInviteDetails> => {
+  return new Promise((res, rej) => {
+    setTimeout(
+      () =>
+        Math.random() > 0.5
+          ? res({
+              project: {
+                name: `Project ${Math.floor(Math.random() * 1000)}`,
+                id: "123",
+              },
+              role: "Project Participant",
+            })
+          : rej(new Error("Could not get invite details")),
+      1000
+    );
+  });
+};
```

Comment:
placeholder for when we have an API endpoint that provides the invite details based on the invite request key/info that's provided